### PR TITLE
Refactored the AssetGenImage code to be another Integration

### PR DESCRIPTION
**Pull request by bramp**
## What does this change?

Refactored the `AssetGenImage` code to be another Integration, so that it looks more similar to SVG/Lottie/Rive/etc. 

This simplifies the assets_generator, and no longer forces `AssetGenImage` to be output even when there are no images.

This can help ensure all integration are on level footing (perhaps helping with task 1 in this list https://github.com/FlutterGen/flutter_gen/issues/384)

## Type of change

- [x] Refactor to cleanup the code base.

## Checklist:

Before submitting your PR, there are a few things you can do to make sure it goes smoothly:

- [ ] Make sure to open a GitHub issue as a bug/feature request before writing your code! That way we can discuss the change, evaluate designs, and agree on the general idea
  - [x] Ensure the tests (`melos run unit:test`)
  - [x] Ensure the analyzer and formatter pass (`melos run format` to automatically apply formatting)

### CODE REVIEW

File: packages/core/lib/generators/integrations/image_integration.dart [hunk 1]
```diff
@@ -0,0 +1,117 @@
+import 'package:flutter_gen_core/generators/integrations/integration.dart';
+import 'package:flutter_gen_core/settings/asset_type.dart';
+
+class ImageIntegration extends Integration {
```
Comment (AlexV525) [inline packages/core/lib/generators/integrations/image_integration.dart:4]: Consider adding docs for the integration.

**Review by AlexV525 — APPROVED** [review]
Looks cool!

File: packages/core/lib/generators/integrations/image_integration.dart (hunk 1 above)
Comment (bramp) [reply packages/core/lib/generators/integrations/image_integration.dart:8]: Added

**Comment by bramp** [comment]
Fixed the merge conflict, and improved the docs. Please take a look.

**Review by AlexV525 — APPROVED** [review]

**Pull request merged**